File: Data Structures & Algorithms/majority-element/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    // majorityElement(nums) {
    //     let counts = new Map();

    //     for (let i = 0; i < nums.length; i++){
    //         if (counts.has(nums[i])){
    //             counts.set(nums[i], counts.get(nums[i])+1);
    //             if (counts.get(nums[i])>nums.length/2){
    //                 return nums[i];
    //             }
    //         } else {
    //             counts.set(nums[i], 1);
    //         }
    //     }

    //     return 0;
    // }
    majorityElement(nums) {
        let candidate = 0;
        let count = 0;

        for (let i = 0; i < nums.length; i++){
            if (count==0){
                candidate=nums[i];
            } 
            count+= (nums[i]==candidate) ? 1 : -1;   
        }

        return candidate;  
    }
}

File: Data Structures & Algorithms/is-subsequence/submission-0.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int l = 0;
        int r = t.length();
        int count = 0;
        for (char c : s.toCharArray()){
            while (l<r){
                if (t.charAt(l)==c){
                    l++;
                    count++;
                    break;
                } else {
                    l++;
                }
            }
        }

        return count==s.length();
    }
}